components/AnchoredHeading.tsx

```diff
@@ -18,15 +18,15 @@ type AnchorHeadingProps = PropsWithChildren<{
  *
  * If your title has some non-English characters, please use
  * `<!---->` to quote your English anchor name inside, with your
- * own title beside it. Otherwises it will return you what it is.
+ * own title beside it. Otherwise it will return you what it is.
  */
 
 // React will ignore "<!" and ">" in the string,
 // so we can just use '-- --' to quote the anchor name inside it.
 const COMMENT_FOR_HEADANCHOR = /--\x20?([\w\x20-]+)\x20?--/;
 
 const AnchoredHeading: FC<AnchorHeadingProps> = ({ level, id, children }) => {
-  const HeadingLevelTag = `h${level}` as any;
+  const HeadingLevelTag = `h${level}` as unknown as JSX.ElementType;
```

Comment:
this is the safe way to cast a string to a HTML element, AFAIK. else TS complains about no overlap